components/tikv_alloc/src/jemalloc.rs

```diff
@@ -0,0 +1,169 @@
+        let res = unsafe { jemallocator::mallctl_set(PROF_DUMP, c_path.get_mut_ptr()) };
+        match res {
+            Err(e) => error!("failed to dump the profile to {:?}: {}", path, e),
+            Ok(_) => {
+                if let Some(p) = path {
+                    info!("dump profile to {}", p);
+                    return;
+                }
+
+                info!("dump profile to {}", env::current_dir().unwrap().display());
+            }
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        extern crate tempdir;
+
+        use self::tempdir::TempDir;
+        use jemallocator;
+        use std::fs;
+
+        const OPT_PROF: &'static [u8] = b"opt.prof\0";
+
+        fn is_profiling_on() -> bool {
+            let mut prof = false;
+            let res = unsafe { jemallocator::mallctl_fetch(OPT_PROF, &mut prof) };
+            match res {
+                Err(e) => {
+                    // Shouldn't be possible since mem-profiling is set
+                    panic!("is_profilng_on: {:?}", e);
+                }
+                Ok(_) => prof,
+            }
+        }
+
+        // Only trigger this test with jemallocs `opt.prof` set to
+        // true ala `MALLOC_CONF="prof:true"`. It can be run by
+        // passing `-- --ignored` to `cargo test -p tikv_alloc`.
+        //
+        // TODO: could probably unignore this by running a second
+        // copy of the executable with MALLOC_CONF set.
+        //
+        // TODO: need a test for the dump_prof(None) case, but
+        // the cleanup afterward is not simple.
+        #[test]
+        #[ignore]
+        fn test_profiling_memory() {
+            // Make sure somebody has turned on profiling
+            assert!(is_profiling_on(), r#"Set MALLOC_CONF="prof:true""#);
+
+            let dir = TempDir::new("test_profiling_memory").unwrap();
+
+            let os_path = dir.path().to_path_buf().join("test1.dump").into_os_string();
+            let path = os_path.into_string().unwrap();
+            super::dump_prof(Some(&path));
+
+            let os_path = dir.path().to_path_buf().join("test2.dump").into_os_string();
+            let path = os_path.into_string().unwrap();
+            super::dump_prof(Some(&path));
+
+            let files = fs::read_dir(dir.path()).unwrap().count();
+            assert_eq!(files, 2);
+
+            // Find the created files and check properties that
+            // indicate they contain something interesting
+            let mut prof_count = 0;
+            for file_entry in fs::read_dir(dir.path()).unwrap() {
+                let file_entry = file_entry.unwrap();
+                let path = file_entry.path().to_str().unwrap().to_owned();
+                if path.contains("test1.dump") || path.contains("test2.dump") {
+                    let metadata = file_entry.metadata().unwrap();
+                    let file_len = metadata.len();
+                    assert!(file_len > 10); // arbitrary number
```

Comment:
According to the "HEAP PROFILE FORMAT" section of https://www.freebsd.org/cgi/man.cgi?query=malloc&sektion=3&manpath=freebsd-release-ports#end we can look for "heap_v2" in the output.

Reply:
@Hoverbear  I do not quite understand you

Reply:
Instead of seeing if the file is larger than 10 bytes, we can look at the content of the file for the substring "heap_v2". This way we are not checking with an arbitrary number.

It's not a blocker.

Reply:
@Hoverbear   This code is not written by me, I don't know much about the context, but I can try to modify it.

Reply:
@Hoverbear  I think the place should be divided into other issues to change it, I currently only have MacOS environment.
 I tried it for a long time and found that the heap profile of `jemalloc` is not supported in MacOS.

Reply:
Yeah, this is for another time.

Reply:
Ok, thanks so much for investigating this @ZhangHanDong !